samples/batch/transcription-enabled-storage/Connector/BatchClient.cs

```diff
@@ -7,67 +7,67 @@ namespace Connector
 {
     using System;
     using System.Collections.Generic;
-    using System.Linq;
     using System.Net;
     using System.Net.Http;
     using System.Text;
+    using System.Threading;
     using System.Threading.Tasks;
     using Connector.Serializable.TranscriptionFiles;
     using Microsoft.Extensions.Logging;
     using Newtonsoft.Json;
 
-    public class BatchClient
+    public static class BatchClient
     {
-        private const string SpeechToTextBasePath = "speechtotext/v3.0/";
+        private const string TranscriptionsBasePath = "speechtotext/v3.0/Transcriptions/";
 
-        private readonly string SubscriptionKey;
+        private static readonly TimeSpan PostTimeout = TimeSpan.FromMinutes(1);
 
-        private readonly string HostName;
+        private static readonly TimeSpan DefaultTimeout = TimeSpan.FromMinutes(2);
 
-        private ILogger Log;
+        private static readonly TimeSpan GetFilesTimeout = TimeSpan.FromMinutes(5);
 
-        public BatchClient(string subscriptionKey, string hostName, ILogger log)
-        {
-            SubscriptionKey = subscriptionKey;
-            HostName = hostName;
-            Log = log;
-        }
+        private static HttpClient HttpClient = new HttpClient() { Timeout = System.Threading.Timeout.InfiniteTimeSpan };
 
         public static bool IsThrottledOrTimeoutStatusCode(HttpStatusCode statusCode)
         {
             if (statusCode == HttpStatusCode.TooManyRequests ||
                 statusCode == HttpStatusCode.GatewayTimeout ||
                 statusCode == HttpStatusCode.RequestTimeout ||
-                statusCode == HttpStatusCode.BadGateway ||
-                statusCode == HttpStatusCode.NotFound)
+                statusCode == HttpStatusCode.BadGateway)
             {
                 return true;
             }
 
             return false;
         }
 
-        public Task<IEnumerable<Transcription>> GetTranscriptionsAsync()
+        public static Task<TranscriptionReportFile> GetTranscriptionReportFileFromSasAsync(string sasUri, ILogger log)
+        {
+            return GetAsync<TranscriptionReportFile>(sasUri, null, DefaultTimeout, log);
+        }
+
+        public static Task<SpeechTranscript> GetSpeechTranscriptFromSasAsync(string sasUri, ILogger log)
         {
-            var path = $"{HostName}{SpeechToTextBasePath}Transcriptions";
-            return this.GetAsync<IEnumerable<Transcription>>(path);
+            return GetAsync<SpeechTranscript>(sasUri, null, DefaultTimeout, log);
         }
 
-        public Task<Transcription> GetTranscriptionAsync(Guid id)
+        public static Task<Transcription> GetTranscriptionAsync(string transcriptionLocation, string subscriptionKey, ILogger log)
```

Comment:
Do we really need to pass subscriptionKey? I would prefer to create a client which is associated with subscritionkey, and so that no need to pass subscriptionkey as parameter

Reply:
I was following the Manage connections guidelines (https://docs.microsoft.com/en-gb/azure/azure-functions/manage-connections) to use static clients for http and storage connections. Of course I could have an instance of a BatchClient that references the static http client, but I thought that this might look a bit more messy.

Reply:
ok, that makes sense for Azure Functions.